File: ruby/lib/command/compress/text_encoder.rb
module Command
  module Compress
    class TextEncoder
      def self.encode(prefix_codes, content)
        new(prefix_codes, content).encode
      end

      private attr_reader :prefix_codes, :content

      def initialize(prefix_codes, content)
        @prefix_codes = prefix_codes
        @content = content
      end

      def encode
        encoded_result = ''
        content.each_char { |c| encoded_result << prefix_codes[c] }

        encoded_result
      end
    end
  end
end

File: python/src/compress/text_encoder.py
class TextEncoder:
  def __init__(self, file_content: str, prefix_codes: dict) -> None:
    self.file_content = file_content
    self.prefix_codes = prefix_codes
  
  def encode(self) -> str:
    encoded_str: list[str] = []
    for char in self.file_content:
      encoded_str.append(self.prefix_codes.get(char))
    
    return ''.join(encoded_str)
